File: react/acorn-react-16-tooling/05/src/object-property-values/Person.js
// @flow
import React from 'react';

type Person = {
  name: string,
  age: number
};

type Props = {
  person: Person
};

export default ({ person }: Props) => (
  <section>
    <h3>Person</h3>
    <p><strong>Name: </strong>{person.name}</p>
    <p><strong>Age: </strong>{person.age}</p>
  </section>
);

File: flutter/udemy-sangwookcho-flutter-bloc/dart_class/bin/person.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'dart:convert';

import 'package:equatable/equatable.dart';

class Person extends Equatable {
  final String name;
  final int age;

  Person({
    required this.name,
    required this.age,
  });

  @override
  List<Object> get props => [name, age];

  Person copyWith({
    String? name,
    int? age,
  }) {
    return Person(
      name: name ?? this.name,
      age: age ?? this.age,
    );
  }

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'name': name,
      'age': age,
    };
  }

  factory Person.fromMap(Map<String, dynamic> map) {
    return Person(
      name: map['name'] as String,
      age: map['age'] as int,
    );
  }

  String toJson() => json.encode(toMap());

  factory Person.fromJson(String source) =>
      Person.fromMap(json.decode(source) as Map<String, dynamic>);

  @override
  bool get stringify => true;
}
